pkg/utils/sshutil/sshutil.go

```diff
@@ -38,15 +40,35 @@ func MarshalPrivateKey(privkey *rsa.PrivateKey) []byte {
 	return privateKeyPEM
 }
 
-// helper function to encrypt a plain-text string using
+// UnMarshalPrivateKey is a helper function that unmarshals a PEM
+// bytes to an RSA Private Key
+func UnMarshalPrivateKey(privateKeyPEM []byte) *rsa.PrivateKey {
+	derBlock, _ := pem.Decode(privateKeyPEM)
+	privateKey, err := x509.ParsePKCS1PrivateKey(derBlock.Bytes)
+
+	if err != nil {
+		return nil
+	}
+	return privateKey
+}
+
+// Encrypt is helper function to encrypt a plain-text string using
 // an RSA public key.
-func Encrypt(pubkey *rsa.PublicKey, msg string) ([]byte, error) {
-	return rsa.EncryptPKCS1v15(rand.Reader, pubkey, []byte(msg))
+func Encrypt(hash hash.Hash, pubkey *rsa.PublicKey, msg string) (string, error) {
+	src, err := rsa.EncryptOAEP(hash, rand.Reader, pubkey, []byte(msg), nil)
+
+	return base64.StdEncoding.EncodeToString(src), err
 }
 
-// helper function to encrypt a plain-text string using
+// Decrypt is helper function to encrypt a plain-text string using
 // an RSA public key.
-func Decrypt(privkey *rsa.PrivateKey, secret string) (string, error) {
-	msg, err := rsa.DecryptPKCS1v15(rand.Reader, privkey, []byte(secret))
-	return string(msg), err
+func Decrypt(hash hash.Hash, privkey *rsa.PrivateKey, secret string) (string, error) {
+	decoded, err := base64.StdEncoding.DecodeString(secret)
```

Comment:
from godoc of `encoding/base64`:
> StdEncoding is the standard base64 encoding, as defined in RFC 4648.
> URLEncoding is the alternate base64 encoding defined in RFC 4648. It is typically used in URLs and file names.

I guess there's no difference when encoding RSA key file into PEM format.